timeseries/src/test/scala/com/criteo/cuttle/timeseries/TimeSeriesSpec.scala

```diff
@@ -111,4 +111,55 @@ class TimeSeriesSpec extends FunSuite with TestScheduling {
     assert(validationSiblings.left.get === List(s"Id badJob is used by more than 1 job"),
            "it means that errors messages are bad")
   }
+
+  test("it should create backfills on valid subperiods") {
+    val job1 = Job("job1", hourly(date"2117-03-25T02:00:00Z"))(completed)
+    val job2 = Job("job2", hourly(date"2117-03-24T02:00:00Z"))(completed)
+
+    // hours 1 2 3 4 5 6 7
+    // job1  VVVVVV--VVVVV
+    // job2  --VVVV----VVV
+    // V means that the job was done on that period. Periods are indivisible.
+    // Backfill requested from hour 2 to 7
+    // Backfills on periods overlapping valid calendar periods are automatically resolved when the actual backfill is run.
+    val jobStates = Map[TimeSeriesUtils.TimeSeriesJob, IntervalMap[Instant, JobState]](
```

Comment:
This setup shoud create 3 backfills : 
- 2-4 (job1, job2)
- 5-6 job1
- 6-7 (job1, job2)

Reply:
Indeed, this will be addressed.

Reply:
Done